src/components/form/form.test.tsx

```diff
@@ -0,0 +1,253 @@
+import React from "react";
+import { render, screen, within } from "@testing-library/react";
+
+import {
+  testStyledSystemSpacing,
+  testStyledSystemPadding,
+} from "../../__spec_helper__/__internal__/test-utils";
+import Form from "./form.component";
+import { StyledFormFooter } from "./form.style";
+import Button from "../button";
+import Dialog from "../dialog";
+
+testStyledSystemSpacing((props) => <Form {...props} />);
+
+test("the form footer is not rendered by default", () => {
+  render(<Form />);
+
+  expect(screen.queryByTestId("form-footer")).not.toBeInTheDocument();
+});
+
+test("renders buttons passed as the leftSideButtons prop", () => {
+  render(
+    <Form
+      leftSideButtons={
+        <>
+          <Button>Left1</Button>
+          <Button>Left2</Button>
+        </>
+      }
+      saveButton={<Button>Save</Button>}
+    />
+  );
+
+  const footerButtons = within(screen.getByTestId("form-footer")).getAllByRole(
+    "button"
+  );
+  expect(footerButtons[0]).toHaveTextContent("Left1");
+  expect(footerButtons[1]).toHaveTextContent("Left2");
+  expect(footerButtons[2]).toHaveTextContent("Save");
+});
+
+test("renders the button passed as the saveButton prop in the form footer", () => {
+  render(<Form saveButton={<Button>Custom Save Button</Button>} />);
+
+  const footerButton = within(screen.getByTestId("form-footer")).getByRole(
+    "button"
+  );
+  expect(footerButton).toHaveTextContent("Custom Save Button");
+});
+
+test("renders buttons passed as the rightSideButtons prop", () => {
```

Comment:
**nit-pick (non-blocking):** Same as above, I don't think its necessary to also test `saveButton` here.

Reply:
same reply [as above](https://github.com/Sage/carbon/pull/6929#discussion_r1745058641) 😃